modal-js/src/dict.ts

```diff
@@ -0,0 +1,261 @@
+import {
+  ObjectCreationType,
+  DictEntry,
+  DictContentsRequest,
+} from "../proto/modal_proto/api";
+import type { DeleteOptions, EphemeralOptions, LookupOptions } from "./app";
+import { client } from "./client";
+import { environmentName } from "./config";
+import { InvalidError, KeyError, RequestSizeError } from "./errors";
+import { dumps, loads } from "./pickle";
+import { ClientError } from "nice-grpc";
+
+// From: modal/_object.py
+const ephemeralObjectHeartbeatSleep = 300_000; // 300 seconds
+
+/** Options to configure a `Dict.put()` operation. */
+export type DictPutOptions = {
+  /** Skip adding the key if it already exists. */
+  skipIfExists?: boolean;
+};
+
+// Dict is a distributed dictionary for key-value storage in Modal apps.
+export class Dict {
```

Comment:
Would be neat if we could make Dict (and probably Queue, haven't checked that implementation) a generic for key/values so you can work with it in a type safe manner